master/internal/db/postgres_rp_workspace_bindings.go

```diff
@@ -2,15 +2,95 @@ package db
 
 import (
 	"context"
+	"database/sql"
 
+	"github.com/pkg/errors"
 	"github.com/uptrace/bun"
+
+	"github.com/determined-ai/determined/master/pkg/model"
+	"github.com/determined-ai/determined/proto/pkg/apiv1"
 )
 
 // RemoveRPWorkspaceBindings removes the bindings between workspaceIds and poolName.
 func (db *PgDB) RemoveRPWorkspaceBindings(ctx context.Context,
-	workspaceIds []int32, poolName string) error {
+	workspaceIds []int32, poolName string,
+) error {
 	_, err := Bun().NewDelete().Table("rp_workspace_bindings").Where("workspace_id IN (?)",
 		bun.In(workspaceIds)).Where("pool_name = ?", poolName).Exec(ctx)
 
 	return err
 }
+
+// ReadWorkspacesBoundToRP get the bindings between workspaceIds and the requested resource pool.
+func ReadWorkspacesBoundToRP(
+	poolName string, offset, limit int32,
+) ([]*model.RpWorkspaceBinding, *apiv1.Pagination, error) {
+	var rpWorkspaceBindings []*model.RpWorkspaceBinding
+	query := Bun().NewSelect().Model(&rpWorkspaceBindings).Where("pool_name = ?",
+		poolName)
+
+	pagination, err := runPagedBunQuery(context.TODO(), query, int(offset), int(limit))
+	if err != nil {
+		if errors.Cause(err) == sql.ErrNoRows {
+			return rpWorkspaceBindings, pagination, nil
+		}
+
+		return nil, nil, err
+	}
+
```

Comment:
don't forget to use `Scan()` on the query to execute it and get the results

Reply:
If I understand correctly, the query will be run at [line 84](https://github.com/determined-ai/determined/pull/7031/files#diff-5b415542f8c43305031c56d94628f15cd0152c001c82988baad2d7b485e1b3d8R84) after pagination is added.

Reply:
ah I see, ok that's good then